helper/config.go

```diff
@@ -133,74 +129,51 @@ var GenesisDoc tmTypes.GenesisDoc
 // var RootChain types.Contract
 // var DepositManager types.Contract
 
-// InitHeimdallConfig initializes with viper config (from heimdall configuration)
-func InitHeimdallConfig(homeDir string) {
-	if strings.Compare(homeDir, "") == 0 {
-		// get home dir from viper
-		homeDir = viper.GetString(HomeFlag)
-	}
-
-	// get heimdall config filepath from viper/cobra flag
-	heimdallConfigFilePath := viper.GetString(WithHeimdallConfigFlag)
+// InitHeimdallConfig initializes passed heimdall/tendermint config files
+func InitHeimdallConfig(rootViper *viper.Viper) error {
+	rootDir := rootViper.GetString(flags.FlagHome)
+	configDir := filepath.Join(rootDir, "config")
 
-	// init heimdall with changed config files
-	InitHeimdallConfigWith(homeDir, heimdallConfigFilePath)
-}
-
-// InitHeimdallConfigWith initializes passed heimdall/tendermint config files
-func InitHeimdallConfigWith(homeDir string, heimdallConfigFilePath string) {
-	if strings.Compare(homeDir, "") == 0 {
-		return
+	heimdallConfigFilePath := filepath.Join(configDir, "heimdall-config.toml")
+	if _, err := os.Stat(heimdallConfigFilePath); os.IsNotExist(err) {
+		hc := GetDefaultHeimdallConfig()
+		WriteConfigFile(heimdallConfigFilePath, &hc)
 	}
 
-	if strings.Compare(conf.BorRPCUrl, "") != 0 {
-		return
-	}
+	// create new viper and
+	configViper := viper.New()
+	configViper.SetConfigType("toml")
+	configViper.SetConfigName("heimdall-config")
+	configViper.AddConfigPath(configDir) // set config file explicitly
 
-	configDir := filepath.Join(homeDir, "config")
-
-	heimdallViper := viper.New()
-	if heimdallConfigFilePath == "" {
-		heimdallViper.SetConfigName("heimdall-config") // name of config file (without extension)
-		heimdallViper.AddConfigPath(configDir)         // call multiple times to add many search paths
-	} else {
-		heimdallViper.SetConfigFile(heimdallConfigFilePath) // set config file explicitly
-	}
-
-	err := heimdallViper.ReadInConfig()
+	err := configViper.ReadInConfig()
 	if err != nil { // Handle errors reading the config file
-		log.Fatal(err)
+		return err
 	}
 
-	if err = heimdallViper.UnmarshalExact(&conf); err != nil {
-		log.Fatalln("Unable to unmarshall config", "Error", err)
+	if err = configViper.UnmarshalExact(&conf); err != nil {
+		return fmt.Errorf("Unable to unmarshall config %v", err)
 	}
 
 	if mainRPCClient, err = rpc.Dial(conf.EthRPCUrl); err != nil {
-		log.Fatalln("Unable to dial via ethClient", "URL=", conf.EthRPCUrl, "chain=eth", "Error", err)
+		return fmt.Errorf("Unable to dial via ethClient. URL=%s, chain=eth, error=%v", conf.EthRPCUrl, err)
 	}
 
 	mainChainClient = ethclient.NewClient(mainRPCClient)
 	if maticRPCClient, err = rpc.Dial(conf.BorRPCUrl); err != nil {
-		log.Fatal(err)
+		return err
 	}
 
 	maticClient = ethclient.NewClient(maticRPCClient)
+
 	// Loading genesis doc
 	genDoc, err := tmTypes.GenesisDocFromFile(filepath.Join(configDir, "genesis.json"))
 	if err != nil {
-		log.Fatal(err)
+		return nil
```

Comment:
Is it a typo? We shouldn't ignore this error here

Reply:
We are handling at cmd level - https://github.com/maticnetwork/heimdall/blob/a7384c5eb7affe10f4992c9963521eedd1cbb42e/cmd/heimdalld/cmd/root.go#L83-L85